src/htm/algorithms/Connections.cpp

```diff
@@ -75,7 +77,23 @@ void Connections::unsubscribe(UInt32 token) {
   eventHandlers_.erase(token);
 }
 
-Segment Connections::createSegment(CellIdx cell) {
+Segment Connections::createSegment(const CellIdx cell, 
+	                           const SegmentIdx maxSegmentsPerCell,
+				   const UInt32 iteration) { //TODO move iteration to Connections.iteration_ ?
+
+  //limit number of segmets per cell. If exceeded, remove the least recently used ones.
+  NTA_ASSERT(maxSegmentsPerCell > 0);
+  while (maxSegmentsPerCell > 1 && numSegments(cell) >= maxSegmentsPerCell) {
+    const auto& destroyCandidates = segmentsForCell(cell);
+    const auto compareSegmentsByLRU = [&](const Segment a, const Segment b) {
+      return (dataForSegment(a).lastUsed < dataForSegment(b).lastUsed); };
```

Comment:
@dkeeney do we need to add here the extra check for `if a.lastUsed == b.lastUsed: a < b` for determinism? The same way you did in SP.